### PR TITLE
Change all julialang.org links to https

### PR DESCRIPTION
**Pull request by tkelman**
The first commit here was from running `contrib/add_license_to_files.jl` then
```
sed -i 's|http://julialang.org|https://julialang.org|g' $(git grep -l http://julialang.org)
sed -i 's|http://docs.julialang.org|https://docs.julialang.org|g' $(git grep -l http://docs.julialang.org)
```

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
+[The learning page](https://julialang.org/learning/) has a great list of resources for new and experienced users alike. [This tutorial video](https://www.youtube.com/watch?v=vWkgEddb4-A) is one recommended starting point, as is the "[Invitation to Julia](https://www.youtube.com/watch?v=gQ1y5NUD_RI)" workshop video from JuliaCon 2015  ([slide materials here](https://github.com/dpsanders/invitation_to_julia)). The [Julia documentation](http://docs.Julialang.org/en/latest/) covers the language and core library features, and is [searchable](http://docs.Julialang.org/en/latest/search/). (note: Javascript required).
```
Comment (omus) [inline CONTRIBUTING.md]: Should the host name be changed to lowercase?

**Review by omus — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (tkelman) [inline CONTRIBUTING.md]: good catch, my sed and grep were case-sensitive so missed the later ones on this line

**Review by Sacha0 — APPROVED** [review]
lgtm (caveat: skimmed rapidly)!

**Comment by StefanKarpinski** [comment]
Man, that touched a lot of files.

**Comment by tkelman** [comment]
Yeah, license headers. Linkcheck doesn't touch the header comments so could split those out, the sed wasn't very discriminating.

**Comment by StefanKarpinski** [comment]
No, it's totally fine – they should be updated too.

**Pull request merged**